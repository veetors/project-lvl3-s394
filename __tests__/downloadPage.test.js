/**
 * @jest-environment node
 */

import { promises as fs } from 'fs';
// import fs from 'fs';
import path from 'path';
import os from 'os';
import nock from 'nock';
import axios from 'axios';
import adapter from 'axios/lib/adapters/http';
import loadPage from '../src';

axios.defaults.adatper = adapter;

const host = 'https://hexlet.io';
const fileName = 'hexlet-io.html';
let content;

beforeEach(async () => {
  const contentPath = path.join(__dirname, '__fixtures__', fileName);
  content = await fs.readFile(contentPath, 'utf-8');

  nock(host)
    .get('/')
    .reply(200, content);
});

test('compare fileNames', async () => {
  const currentFileName = await loadPage(host, os.tmpdir());

  expect(currentFileName).toBe(fileName);
});

test('compare content', async () => {
  const currentFileName = await loadPage(host, os.tmpdir());
  const fileContent = await fs.readFile(path.join(os.tmpdir(), currentFileName), 'utf-8');

  expect(content).toBe(fileContent);
});
