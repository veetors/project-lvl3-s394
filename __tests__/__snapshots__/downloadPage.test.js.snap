// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error bad url 1`] = `"Request failed with status code 404"`;

exports[`error output directory does not exist 1`] = `"ENOENT: no such file or directory, mkdir '/ENOENT/hexlet-io-courses_files'"`;

exports[`error permission denied 1`] = `"EACCES: permission denied, mkdir '/hexlet-io-courses_files'"`;
